client/state/sites/test/selectors.js

```diff
@@ -3642,4 +3643,42 @@ describe( 'selectors', () => {
 			).toBe( true );
 		} );
 	} );
+
+	describe( 'isSiteOnWooExpressTrialPeriod()', () => {
+		test( 'Should return true when the e-commerce trial is in the purchases list', () => {
+			const initialPurchases = Object.freeze( [
+				{ ID: 1, product_slug: PLAN_ECOMMERCE_TRIAL_MONTHLY, blog_id: 1337 },
+			] );
+
+			const state = {
+				purchases: {
+					data: initialPurchases,
+					error: null,
+					isFetchingSitePurchases: false,
+					isFetchingUserPurchases: false,
+					hasLoadedSitePurchasesFromServer: false,
+					hasLoadedUserPurchasesFromServer: true,
+				},
+			};
+
+			expect( isSiteOnWooExpressTrialPeriod( state ) ).toBeTruthy();
+		} );
+
+		test( 'Should return false when the e-commerce trial is not in the purchases list', () => {
```

Comment:
It might be worth adding test cases where we have a non-plan purchase and a plan purchase that isn't the eCommerce trial -- the main eCommerce plan might be the best way to test that.